lib/private/Authentication/Login/FlowV2EphemeralSessionsCommand.php

```diff
@@ -10,15 +10,18 @@
 
 use OC\Core\Controller\ClientFlowLoginV2Controller;
 use OCP\ISession;
+use OCP\IURLGenerator;
 
 class FlowV2EphemeralSessionsCommand extends ALoginCommand {
 	public function __construct(
 		private ISession $session,
+		private IURLGenerator $urlGenerator,
 	) {
 	}
 
 	public function process(LoginData $loginData): LoginResult {
-		if (str_starts_with($loginData->getRedirectUrl() ?? '', '/login/v2/grant')) {
+		// Looking for: "/login/v2/grant?user=&direct=0&stateToken=$token" or "/subfolder/login/v2/grant?user=&direct=0&stateToken=$token"
+		if (str_starts_with($loginData->getRedirectUrl() ?? '', $this->urlGenerator->getWebroot() . '/login/v2/grant')) {
```

Comment:
Does it not work with:

```suggestion
		if (str_starts_with($loginData->getRedirectUrl() ?? '', $this->urlGenerator->linkToRoute('core.ClientFlowLoginV2')) {
```

Reply:
Much cleaner :+1: